HHS/Head-Start-TTADP
# Move report table to own component

## Description of change

 * Report table is now it's own component
 * Activity reports table replaced with new stand alone component
 * My alerts still needs to be updated to use new component. There are a few difference between the two tables that need to be handled. Saving for a later date to keep the change smaller
 * `withWidgetData` uses deep compare to prevent multiple renders
 * Moved table related tests from landing page to new component
 * Add report table to grantee dashboard

Only added functionality is the ability to show or hide the filter menu on report tables, and dynamically setting the table caption.

## How to test

View activity reports table on landing page. Check the following:
 * filters, changing regions
 * filters are preserved through region changes and regions through filter changes
 * correct results are displayed when filtering
 * download buttons still function
 * you can select individual and all reports
 * sorting still works
 * paging still works

View reports on the grantee TTA record page. Check the same as above. There won't be filter controls and the reports should be scoped to the current grantee.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-396

## Notes

This PR will wait for global filters to show up on the grantee TTA page before merging.

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
